File: Week1/flamion/ModifiedFizzBuzz.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int m = scanner.nextInt();

        Map<Integer, String> replaceLookup = new HashMap<>();

        for (int i = 0; i < m; i++) {
            replaceLookup.put(scanner.nextInt(), scanner.next());
        }

        StringBuilder builder = new StringBuilder();

        for (int i = 1; i <= n; i++) {
            for (int j : replaceLookup.keySet()) {
                if (i % j == 0) {
                    builder.append(replaceLookup.get(j));
                }
            }
            System.out.println(builder.isEmpty() ? i : builder);
            builder.setLength(0);
        }
    }
}

File: Week1/flamion/ModifiedFizzBuzz.kt
import java.util.*

fun main() {
    val scanner = Scanner(System.`in`)

    val n = scanner.nextInt()
    val m = scanner.nextInt()

    val replaceLookup: MutableMap<Int, String> = java.util.HashMap()

    for (i in 0 until m) {
        replaceLookup[scanner.nextInt()] = scanner.next()
    }

    val builder = StringBuilder()

    for (i in 1..n) {
        for (j in replaceLookup.keys) {
            if (i % j == 0) {
                builder.append(replaceLookup[j])
            }
        }
        println(if (builder.isEmpty()) i else builder)
        builder.setLength(0)
    }
}
